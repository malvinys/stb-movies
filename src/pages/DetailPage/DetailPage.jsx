import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { LazyLoadImage } from 'react-lazy-load-image-component';

import { fetchDetailMovie, moviesSelector } from '../../models/Movies';

const DetailPage = () => {
  const { id } = useParams();
  const dispatch = useDispatch();
  const { movie } = useSelector(moviesSelector);
  const [backdrop, setBackdrop] = useState();
  const [image, setImage] = useState();
  const [title, setTitle] = useState();
  const [releaseYear, setReleaseYear] = useState();
  const [rating, setRating] = useState();
  const [overview, setOverview] = useState();

  useEffect(() => {
    dispatch(fetchDetailMovie(id));
  }, [dispatch]);

  useEffect(() => {
    if (movie) {
      const backdropUrl = movie.Poster;
      const imageUrl = movie.Poster;
      const releseYear = movie.Year;

      setBackdrop(backdropUrl);
      setImage(imageUrl);
      setTitle(movie.Title);
      setReleaseYear(releseYear);
      setRating(movie.imdbRating);
      setOverview(movie.Plot);
    }
  }, [movie]);

  return (
    <>
      <section className="base-section section-movie-detail" style={{ backgroundImage: `linear-gradient(to right, rgB(0,0,0, 0.8) 0%, rgB(0,0,0, 0.6) 100%), url("${backdrop}")` }}>
        {image && <LazyLoadImage className="section-movie-detail-image" src={image} alt={title} effect="blur" />}
        {title && (
          <div className="section-movie-detail-content">
            <h2>
              {title}
              <span>
                (
                {releaseYear}
                )
              </span>
            </h2>
            <div className="section-movie-detail-content-rating">{rating}</div>
            <p>
              Overview
              <br />
              <span>{overview}</span>
            </p>
          </div>
        )}
      </section>
    </>
  );
};

export default DetailPage;
