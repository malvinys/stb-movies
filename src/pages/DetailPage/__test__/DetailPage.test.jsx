import * as React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import { Provider } from 'react-redux';

import store from '../../../models/_store';

import DetailPage from '../DetailPage';

const shallowRenderer = createRenderer();

describe('<DetailPage />', () => {
  it('should match snapshot', () => {
    shallowRenderer.render(
      <Provider store={store}>
        <DetailPage />
      </Provider>,
    );
    const renderedOutput = shallowRenderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
