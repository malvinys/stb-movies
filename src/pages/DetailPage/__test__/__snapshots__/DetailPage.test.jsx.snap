// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailPage /> should match snapshot 1`] = `
<Context.Provider
  value={
    Object {
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <DetailPage />
</Context.Provider>
`;
